src/zarr/core/metadata.py

```diff
@@ -602,9 +602,21 @@ def parse_fill_value_v3(fill_value: Any, dtype: FLOAT_DTYPE) -> FLOAT: ...
 def parse_fill_value_v3(fill_value: Any, dtype: COMPLEX_DTYPE) -> COMPLEX: ...
 
 
+@overload
+def parse_fill_value_v3(fill_value: Any, dtype: np.dtype[Any]) -> Any:
```

Comment:
```suggestion
def parse_fill_value_v3(fill_value: Any, dtype: np.dtype[Any]) -> np.generic:
```
To indicate that a scalar is always returned? (like the other narrower types, which are e.g., np.uint8 which represents a single scalar I think?)

Reply:
That passes. The revealed type of

```
reveal_type(parse_fill_value_v3(1, np.dtype("uint64")))
```

is `np.generic`, which sounds better than `Any`.